### PR TITLE
Update GitHub actions dependencies

### PR DESCRIPTION
**Pull request by acelaya**
This pull request addresses a couple of warnings printed in our workflows, by updating to the latest instances of some actions and changing how we pass info between jobs via output.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
         id: list-python-versions
         run: |
           pip install hdev
-          echo "::set-output name=python_headline_version::`hdev python_version --style plain --first`"
+          echo "python_headline_version=`hdev python_version --style plain --first`" >> $GITHUB_OUTPUT
```
Comment (acelaya) [inline .github/workflows/ci.yml:37]: See https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

**Comment by robertknight** [comment]
Just a note that I would abbreviate GitHub Actions as GHA, to avoid confusion with Google Analytics, which I believe is still used by some projects, though not the client any more.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -23,7 +23,7 @@ jobs:
 
     steps:
       - name: Checkout code
-        uses: actions/checkout@v1
+        uses: actions/checkout@v3
 
       - name: Setup python
         uses: actions/setup-python@v2
```
Comment (robertknight) [inline .github/workflows/ci.yml:29]: I notice we're getting some warnings related to `actions/setup-python@v2`. Those can be addressed separately if that's easier.

> Node.js 12 actions are deprecated. Please update the following actions to use Node.js 16: actions/setup-python@v2. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/.

**Review by robertknight — APPROVED** [review]
LGTM. There are CI warnings related to use of an old version of Node by `actions/setup-python@v2`, but those can be addressed separately.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (acelaya) [reply .github/workflows/ci.yml:29]: Yep, I wanted to take those separately, because latest version was v4 and I still need to go over the changelog.

**Pull request merged**